### PR TITLE
Fix #455: Filter out empty tags

### LINKED ISSUE #455: When pushing a post, we should trim "empty" (full of whitespaces) tags
It's a problem with XMLRPC because it can prevent post upload when the user enter a wrong tag:

```
05-30 08:01:41.579 1785-1785/org.wordpress.android E/WordPress-API: Volley error on https://bia.is/wp2/xmlrpc.php
  com.android.volley.VolleyError: org.wordpress.android.fluxc.network.xmlrpc.XMLRPCFault: A name is required for this term.
      at org.wordpress.android.fluxc.network.xmlrpc.XMLRPCRequest.parseNetworkResponse(XMLRPCRequest.java:63)
      at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:125)
```

We should strip trim tags and remove empty ones before we send the PUSH_POST action.

### PR DESCRIPTION
**Pull request by maxme**
Fixes #455 by removing whitespaces-only tags from the list of tags attached to post upload.

Note: tags are trimmed by the server when we push them, so it's safe to trim them in the client.

### CODE REVIEW

**Comment by maxme** [comment]
We should use this in wpandroid 7.5.

File: example/src/test/java/org/wordpress/android/fluxc/post/PostModelTest.java [hunk 1]
```diff
@@ -110,4 +110,28 @@ public void testLocation() {
         assertTrue(testPost.shouldDeleteLatitude());
         assertTrue(testPost.shouldDeleteLongitude());
     }
+
+    @Test
+    public void testFilterEmtpyTagsOnGetTagNameList() {
```
Comment (aforcier) [inline example/src/test/java/org/wordpress/android/fluxc/post/PostModelTest.java]: s/Emtpy/Empty

Comment (maxme) [inline example/src/test/java/org/wordpress/android/fluxc/post/PostModelTest.java]: Updated in  	0279349, next step: add a method name spell checking configuration in checkstyle.

**Comment by aforcier** [comment]
:shipit:

**Pull request merged**